dlib/cuda/cudnn_dlibapi.cpp

```diff
@@ -162,6 +162,12 @@ namespace dlib
             return des.get_handle();
         }
 
+        static cudnnActivationDescriptor_t mish_activation_descriptor()
+        {
+            thread_local cudnn_activation_descriptor des(CUDNN_ACTIVATION_MISH, CUDNN_PROPAGATE_NAN,0);
```

Comment:
This isn't available in all versions of cudnn. There should be some `#if` preprocessor check around this to allow old versions of cudnn to be used.  The simplest thing to do is to provide a CUDA method that implements this stuff and call that if cudnn is too old.  You can use a grid stride loop to do this easily by basically just copying and pasting your C++ code into the cuda kernel.  For example, this is the cuda code in dlib that implements prelu:

```
        __global__ void _cuda_prelu(const float* s, float* d, size_t n, const float* pp)
        {
            const float p = *pp; 
            for (auto i : grid_stride_range(0, n))
            {
                if (s[i] > 0) 
                    d[i] = s[i];
                else
                    d[i] = p*s[i];
            }
        }

        void prelu (
            tensor& dest,
            const tensor& src, 
            const tensor& param
        )
        {
            launch_kernel(_cuda_prelu, max_jobs(dest.size()), 
                src.device(), dest.device(), src.size(), param.device());
        }
```
Nothing tricky is required. Honestly it's simpler than calling a lot of the cuDNN functions.